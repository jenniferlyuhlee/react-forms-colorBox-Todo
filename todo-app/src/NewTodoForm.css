.NewTodoForm{
    display: flex;
    align-items: center;
    margin-bottom: 20px;
}

.todo-input{
    margin: 0 8px;
}