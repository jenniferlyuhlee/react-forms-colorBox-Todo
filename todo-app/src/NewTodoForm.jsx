import React, { useState } from "react";
import "./NewTodoForm.css"

function NewTodoForm({addTodo}) {
    const [task, setTask] = useState("");

    const handleChange = e => {
        setTask(e.target.value)
    }

    const handleSubmit = e => {
        e.preventDefault();
        addTodo({task})
        setTask("");
    }

    return(
        <form className = "NewTodoForm" onSubmit = {handleSubmit}>
            <label className = "todo-label" htmlFor="task">I need to: </label>
            <input className = "todo-input"
                   id = "task" 
                   type = "text"
                   name = "task"
                   placeholder = "Ex: Wash dishes"
                   value = {task}
                   onChange = {handleChange}/>
            <button className ="NewTodoForm-btn">Add!</button>
        </form>
    )
}

export default NewTodoForm;