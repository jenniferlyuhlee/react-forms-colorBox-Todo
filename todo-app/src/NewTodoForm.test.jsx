import React from "react";
import {fireEvent, render} from "@testing-library/react"
import NewTodoForm from "./NewTodoForm";

// smoke test
test("it renders without crashing", () => {
    render(<NewTodoForm />)
})

// snapshot test
test("it matches snapshot", () => {
    const {asFragment} = render(<NewTodoForm />)
    expect(asFragment()).toMatchSnapshot();
})