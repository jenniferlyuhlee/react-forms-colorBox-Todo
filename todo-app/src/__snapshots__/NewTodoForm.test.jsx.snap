// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <label
      class="todo-label"
      for="task"
    >
      I need to: 
    </label>
    <input
      class="todo-input"
      id="task"
      name="task"
      placeholder="Ex: Wash dishes"
      type="text"
      value=""
    />
    <button
      class="NewTodoForm-btn"
    >
      Add!
    </button>
  </form>
</DocumentFragment>
`;
